src/transformers/models/mega/modeling_mega.py

```
@@ -0,0 +1,2299 @@
+
+        padding_mask: tensor with shape (batch x no. chunks x chunk size x sequence length), or None causal_mask:
+        tensor with shape (sequence length x sequence length), or None
+
+        Both masks expect 1 to indicate *not masked* and 0 to indicate *masked*
+        """
+        slen = k.size(2)
+        if padding_mask is not None:
+            # 1 for *not masked*
+            # 0 for *masked*
+
+            # B x K x 1
+            lengths = padding_mask.sum(-1, keepdim=True)
+            # B x K x 1 x 1
+            lengths = lengths.clamp(min=1.0).unsqueeze(-1)
+        else:
+            lengths = slen
+
+        if causal_mask is not None:
+            lengths = causal_mask.sum(dim=-1, keepdim=True)
+
+        # C x C
+        bias = self.rel_pos_bias(slen)
+        if slen != q.size(2):
+            assert q.size(2) == 1, "Size mismatch between Q and K in element attention"
+            # 1 x C
+            bias = bias[-1:]
+
+        # B x K x C x C
+        qk = torch.matmul(q, k.transpose(2, 3)) / lengths + bias
+
+        if self.config.attention_activation == "relu2":
+            attn_weights = relu2(qk).type_as(qk)
+        elif self.config.attention_activation == "laplace":
+            attn_weights = laplace(qk).type_as(qk)
+        else:
+            raise ValueError(f"Unknown attention activation function: {self.config.attention_activation}")
+
+        if padding_mask is not None:
+            attn_weights = attn_weights * padding_mask.unsqueeze(2)
+
+        if causal_mask is not None:
+            attn_weights = attn_weights * causal_mask
+
+        return attn_weights
+
+    def softmax_attention(self, q, k, padding_mask, causal_mask):
+        "Standard softmax attention with combined padding/attention mask"
+        slen = k.size(2)
+        # C x C
+        bias = self.rel_pos_bias(slen)
+        if slen != q.size(2):
+            assert q.size(2) == 1, "Size mismatch between Q and K in softmax attention"
```

no assedrt, let's raise an error